conpot/protocols/guardian_ast/guardian_ast_server.py

```diff
@@ -0,0 +1,316 @@
+# Copyright (C) 2015  Lukas Rist <glaslos@gmail.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+Service support based on gaspot.py [https://github.com/sjhilt/GasPot]
+Original authors: Kyle Wilhoit and Stephen Hilt
+"""
+
+import gevent
+from gevent.server import StreamServer
+
+import datetime
+import random
+
+import logging as logger
+
+import conpot.core as conpot_core
+
+
+class GuardianASTServer(object):
+    def __init__(self, template, template_directory, args):
+        self.server = None
+        self.databus = conpot_core.get_databus()
+        # dom = etree.parse(template)
+        self.fill_offset_time = datetime.datetime.utcnow()
+        logger.info('Conpot GuardianAST initialized')
+
+    def handle(self, sock, addr):
+        session = conpot_core.get_session('guardian_ast', addr[0], addr[1])
+        logger.info('New GuardianAST connection from %s:%d. (%s)', addr[0], addr[1], session.id)
+        session.add_event({'type': 'NEW_CONNECTION'})
+        current_time = datetime.datetime.utcnow()
+        fill_start = self.fill_offset_time - datetime.timedelta(minutes=313)
```

Comment:
Yeah, there are a couple of diff values in there that might belong into the config (or randomized too). I'd leave that for the next round...